### PR TITLE
Remake the built-in color schemes to modern standards

### PR DESCRIPTION
**Pull request by romainl**
- replace the original color schemes with the remakes
- improve check_colors.vim
- rewrite check_colors.vim in vim9script
- update CODEOWNERS
- update `test_gui.vim` to match new `torte.vim`

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -19,6 +19,26 @@ runtime/autoload/tar.vim		@cecamp
...
+runtime/colors/industry.vim		@habamax @romainl @neutaaaaan
+runtime/colors/koehler.vim		@habamax @romainl @neutaaaaan
+runtime/colors/morning.vim		@habamax @romainl @neutaaaaan
+runtime/colors/murphy.vim		@habamax @romainl @neutaaaaan
+runtime/colors/pablo.vim		@habamax @romainl @neutaaaaan
+runtime/colors/peachpuff.vim		@habamax @romainl @neutaaaaan
+runtime/colors/ron.vim			@habamax @romainl @neutaaaaan
+runtime/colors/shine.vim		@habamax @romainl @neutaaaaan
+runtime/colors/slate.vim		@habamax @romainl @neutaaaaan
+runtime/colors/tools			@habamax @romainl @neutaaaaan
```
Comment (k-takata) [inline .github/CODEOWNERS:38]: Maybe one of line 22 or line 39 can be removed.

**Comment by habamax** [comment]
@chrisbra 

```
Failures: 
	From test_gui.vim:
	Found errors in Test_colorscheme():
	command line..script C:/projects/vim/src/testdir/runtest.vim[456]..function RunTheTest[44]..Test_colorscheme line 19: Pattern '\nSearch         xxx term=reverse ctermfg=0 ctermbg=12 gui=bold guifg=Black guibg=Red' does not match '\nSearch         xxx term=reverse cterm=reverse ctermfg=196 ctermbg=16 gui=reverse guifg=#ff0000 guibg=#000000'
```

Do you think this test is needed? Given the fact it could be changed in the future for `torte.vim`?

We can fix the test to match what we have now for torte, but ...

**Comment by romainl** [comment]
@brammool @chrisbra this is the first PR related to the colorscheme overhaul we have discussed before in #1665 and #4996. So far, it contains:

- the  remakes of the 17 bundled colorschemes (not `default`),
- an updated `runtime/colors/tools/check_colors.vim`, rewritten in vim9script,
- an updated `.github/CODEOWNERS`.

We have noticed an issue in `test_gui.vim`, that should be fixed in the test suite itself:

    Failures: 
	From test_gui.vim:
	Found errors in Test_colorscheme():
	command line..script C:/projects/vim/src/testdir/runtest.vim[456]..function RunTheTest[44]..Test_colorscheme line 19: Pattern '\nSearch         xxx term=reverse ctermfg=0 ctermbg=12 gui=bold guifg=Black guibg=Red' does not match '\nSearch         xxx term=reverse cterm=reverse ctermfg=196 ctermbg=16 gui=reverse guifg=#ff0000 guibg=#000000'

and there might be some discrepancies in the various places where the whole "colorscheme" business is documented.

File: .github/CODEOWNERS (hunk 1 above)
Comment (romainl) [reply .github/CODEOWNERS:38]: Fixed in #[236442a](https://github.com/vim/vim/pull/9795/commits/236442aeac91f56326ff5164713dd2569ccc792c).

**Comment by romainl** [comment]
> @chrisbra
> 
> ```
> Failures: 
> 	From test_gui.vim:
> 	Found errors in Test_colorscheme():
> 	command line..script C:/projects/vim/src/testdir/runtest.vim[456]..function RunTheTest[44]..Test_colorscheme line 19: Pattern '\nSearch         xxx term=reverse ctermfg=0 ctermbg=12 gui=bold guifg=Black guibg=Red' does not match '\nSearch         xxx term=reverse cterm=reverse ctermfg=196 ctermbg=16 gui=reverse guifg=#ff0000 guibg=#000000'
> ```
> 
> Do you think this test is needed? Given the fact it could be changed in the future for `torte.vim`?
> 
> We can fix the test to match what we have now for torte, but ...

I went ahead and fixed that part of `test_gui.vim`.

**Comment by brammool** [comment]
Thanks for doing all this work!

Let's keep this PR open for at least a few days, so that people can comment.
Please update the PR for any reported problems.
If there is a discussion about choices, you might point to a discussion that took place in the repository where this is coming from.

File: runtime/colors/tools/check_colors.vim [hunk 2]
```diff
@@ -1,238 +1,209 @@
...
-" Load the scheme and source this script, like this:
-"    :edit colors/desert.vim | :so colors/tools/check_colors.vim
+vim9script
+# This script tests a color scheme for some errors and lists potential errors.
+# Load the scheme and source this script, like this:
+#    :edit colors/desert.vim | :ru colors/tools/check_colors.vim
 
-let s:save_cpo= &cpo
+var save_cpo = &cpo
 set cpo&vim
```
Comment (bfrg) [inline runtime/colors/tools/check_colors.vim]: You don't need to save, set and restore `cpoptions` in a `vim9script` file. See the third paragraph under `:h E1101`.

**Comment by romainl** [comment]
Please read these two threads which provide most of the context around this PR:

- https://github.com/vim/vim/issues/1665
- https://github.com/vim/vim/issues/4996

Discussions regarding specific aspects of specific colorschemes are better suited for the issue tracker at  https://github.com/vim/colorschemes, where all the work that went into this PR actually happened.

File: runtime/colors/tools/check_colors.vim [hunk 3]
```diff
@@ -1,238 +1,209 @@
...
 
-func! Test_check_colors()
-  let l:savedview = winsaveview()
-  call cursor(1,1)
-  let err = {}
+def Test_check_colors()
+    var savedview = winsaveview()
```
Comment (bfrg) [inline runtime/colors/tools/check_colors.vim]: Variables that don't change can be declared constant. Vim supported this even before `vim9script`.

Writing
```vim
const savedview = winsaveview()
```
shows the clear intention of the variable. Plus it catches errors early. It protects you from accidentally changing variables that aren't intended to be changed. In some languages it's called _const-correctness_.

Same can be done for the variables `hi_groups`, `bg_set`, `bg_let`, `bg_pat`, `line`  below.

Comment (habamax) [reply runtime/colors/tools/check_colors.vim]: I have fixed it in vim/colorschemes repo, I think @romainl will add it to PR.

File: runtime/colors/tools/check_colors.vim (hunk 2 above)
Comment (romainl) [reply runtime/colors/tools/check_colors.vim]: Fixed, thank you.

File: runtime/colors/tools/check_colors.vim (hunk 3 above)
Comment (romainl) [reply runtime/colors/tools/check_colors.vim]: Fixed. Could you check?

**Review by bfrg — APPROVED** [review]

**Comment by vitaly-zdanevich** [comment]
Please attach screenshots, maybe before/after.

**Comment by romainl** [comment]
> Please attach screenshots, maybe before/after.

Will do today, but in a page on our wiki to keep this thread manageable.

**Comment by romainl** [comment]
Here is a sampler I am working on: https://github.com/vim/colorschemes/wiki/Remake-sampler-%E2%80%94-256-colors. It is not complete yet but it should give an idea of what we have been doing.

**Comment by habamax** [comment]
@romainl could you update PR with fixes we did wrt comments we've got on our issue tracker?

**Comment by romainl** [comment]
> @romainl could you update PR with fixes we did wrt comments we've got on our issue tracker?

Done.

**Comment by Shane-XB-Qian** [comment]
this would not (or not plan to) change/impact the default (perhaps can be treated as a special 'builtin' one) specially when/with 'not termguicolors', right ?
// in server env, or personally perhaps users did not like change that default one.

**Comment by habamax** [comment]
>this would not (or not plan to) change/impact the default

default colorscheme is untouched.

**Comment by romainl** [comment]
> this would not (or not plan to) change/impact the default

We would actually like to improve `default`, which has lots of the same issues the others have, but it is not actually implemented the same way the other original colorschemes are (so templating, as we used here, is out of the question) and, being "default", may require extra care, therefore it is not going to happen any time soon.

**Comment by romainl** [comment]
The last commit addresses various problems encountered in the Windows console with `bold`, `underline`, and `reverse`. I believe this is how far we can go for that environment without fixing the long-lasting low-level issues with terminal capabilities detection.

**Comment by romainl** [comment]
The last commit fixes legibility issues with `DiffText` in all colorschemes as well as a relatively uncommon bug regarding `TabLineSel` in several of them.

@chrisbra @brammool What do you think?

**Comment by romainl** [comment]
@brammool as discussed, we applied the last missing changes and slightly updated `colors/README.txt` so we consider this feature to be ready.

**Comment by brammool** [comment]
The only way to find out if users generally like this is to include it.    So let me include it, then we can discuss any further changes.  Or roll it back if there are too many complaints.

**Comment by brammool** [comment]
I have included this and I'll send out a runtime files updte.

**Pull request closed without merging**